src/redisearch_rs/redis_mock/src/allocator.rs

```diff
@@ -5,16 +5,19 @@
  * Licensed under your choice of the Redis Source Available License 2.0
  * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
  * GNU Affero General Public License v3 (AGPLv3).
-*/
+ */
 
-//! Alternative implementations of Redis' functions for heap allocation.
-use std::{
-    alloc::{Layout, alloc, alloc_zeroed, dealloc, realloc},
-    ffi::c_void,
-};
+use std::alloc::{Layout, alloc, alloc_zeroed, dealloc, realloc};
+use std::ffi::c_void;
+use std::ptr;
 
-const HEADER_SIZE: usize = std::mem::size_of::<usize>();
-const ALIGNMENT: usize = std::mem::align_of::<usize>();
+const ALIGNMENT: usize = 2 * std::mem::align_of::<usize>();
```

Comment:
Is this platform specific or are we convinced this will be OK for all platforms?

Reply:
yes, it's to match `max_align_t`, this equation works for 32-bit and 64-bit alike